src/profiles_management/helpers/kfam.py

```diff
@@ -102,5 +212,96 @@ def list_contributor_authorization_policies(
     return [
         ap
         for ap in authorization_policies
-        if has_kfam_annotations(ap) and not resource_is_for_profile_owner(ap)
+        if has_valid_kfam_annotations(ap) and not resource_is_for_profile_owner(ap)
     ]
+
+
+def kfam_resources_list_to_roles_dict(
+    resources: List[RoleBinding] | List[GenericNamespacedResource],
+) -> dict[str, List[ContributorRole]]:
+    """Convert list of KFAM RoleBindings or AuthorizationPolicies to dict.
+
+    The user of the resource will be used as a key and its role as the value.
+
+    Args:
+        resources: List of KFAM RoleBindings or AuthorizationPolicies.
+
+    Returns:
+        Dictionary with keys the user names and values the roles, derived from parsing all
+        the provided resources.
+    """
+    contributor_roles_dict = {}
+    for resource in resources:
+        if has_valid_kfam_annotations(resource):
+            user = get_contributor_user(resource)
+            role = get_contributor_role(resource)
+            contributor_roles_dict[user] = contributor_roles_dict.get(user, []) + [role]
```

Comment:
`get_contributor_*()` is already doing a check for `has_valid_kfam_annotations()`. You can either avoid doing the check here again or modify the method and do the check here, either way works for me. Please note if you pick the latter solution, you'll probably have to update it also whenever `get_contributor_*()` is called.

Reply:
plus, the calling function was using `list_contributor_rolebindings`, so the objects passed will always have the KFAM annotations.

dff9d581f86c1f79ae3b6328e72aeca62642ecbe